### PR TITLE
add regression coverage for Issue 22743

### PR DESCRIPTION
**Pull request by labkey-chrisj**
#### Rationale
This adds test coverage for Issue 22743, which is essentially that there isn't adequate/comprehensive test coverage for file and attachment columns in various domain kinds

#### Related Pull Requests
n/a

#### Changes

- [x] test for list, samples, assay, dataset with files/attachments, export and import
- [x] test for sample file fields, ui
- [x] test for assay run and result domain file fields, ui

### CODE REVIEW

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 1]
```diff
@@ -15,153 +15,609 @@
...
+        {
+            var queryUpdatePage = samplesRegion.clickInsertNewRow();
+            queryUpdatePage.setField("Name", file.getName());
+            queryUpdatePage.setField("file", file)
+                    .submit();
+        }
+        validateSampleData(SUBA_SAMPLETYPE_NAME, SUBFOLDER_A_PATH, SAMPLE_FILES);
+
+        var fileContentPage = FileContentPage.beginAt(this, SUBFOLDER_A_PATH);
+        fileContentPage.fileBrowserHelper().selectFileBrowserItem("/sampletype/csv_sample.csv");
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java:222]: why "csv_sample.csv" needs to selectFileBrowserItem?

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 2]
```diff
@@ -15,153 +15,609 @@
...
+    @Test
+    public void testSampleFileFields()
     {
-        _containerHelper.deleteProject(getProjectName(), afterTest);
+        SampleTypeDefinition subASampleType = new SampleTypeDefinition(SUBA_SAMPLETYPE_NAME)
+                .setFields(List.of(new FieldDefinition("color", ColumnType.String),
+                        new FieldDefinition("file", ColumnType.File)));
+        SampleTypeAPIHelper.createEmptySampleType(SUBFOLDER_A_PATH, subASampleType);
+
+        // give the sampleType actual files by adding individual rows; for the nonce not all files are resolving on import
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Could you explain more why/what is not resolving on import?

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 3]
```diff
@@ -15,153 +15,609 @@
...
-        init.doSetup();
+        String runName = "assay_run.tsv";
+
+        // generate some resultFiles
+        var resultFiles = new ArrayList<File>();
+        WebDavUploadHelper uploadHelper = new WebDavUploadHelper(SUBFOLDER_A_PATH);
+        List<Map<String, Object>> importData = new ArrayList<>();
+        for (int i = 0; i < 5; i++)
+        {
+            String fileName = String.format("field_file_for_results_domain-%d.tsv", i);
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Why using 5 different result files? This seems they could be 5 data rows in a single result file. Or, have 1 result file that has N rows, and another result file with 1 row.

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 4]
```diff
@@ -15,153 +15,609 @@
...
+            uploadHelper.uploadFile(resultFile);
+            resultFiles.add(resultFile);
+
+            importData.add(Map.of(RESULT_TXT_COL, result, RESULT_FILE_COL, resultFile.getName(),
+                    OTHER_RESULT_FILE_COL, otherResultFile.getName(),
+                    RUN_TXT_COL,"run text", RUN_FILE_COL, "runFile.tsv"));
+        }
+
+        // generate a run file, referencing the result files
+        String importDataFileContents = TestDataUtils.tsvStringFromRowMaps(importData,
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Could use any file for run file, the content doesn't seem to matter, but using an existing static file should simplify the test

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 5]
```diff
@@ -15,153 +15,609 @@
...
+        beginAt(SUBFOLDER_A_PATH + "/project-begin.view");
+        goToModule("Assay");
+        clickAndWait(Locator.linkContainingText(SUB_A_ASSAY));
+        clickButton("Import Data");
+        clickButton("Next");    // batch properties
+
+        // run properties
+        setFormElement(Locator.input("name"), runName);
+        setFormElement(Locator.input(RUN_TXT_COL), "run text");
+        setFormElement(Locator.input(RUN_FILE_COL), importFile);
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Could use any existing sample file for importFile. Also, could you please rename this to runFileFieldFile?

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java (hunk 2 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Ah.  That *was* true when that code was written- your fix in platform resolved the tricky-characters pdf file import that wasn't working in the sampleType in export folder.  I'll update

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java (hunk 1 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/FileAttachmentColumnTest.java:222]: selecting it with this path was the only way I could get the file browser to open into the /sampleType container.
It was just a file I knew I could expect to be there

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java (hunk 3 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/FileAttachmentColumnTest.java]: It was just the way that occurred for me to do it at the time.
If you would like me to change this, I will.

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java (hunk 2 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Come to think of it, now that the samples are importing correctly we may not need this test at all- I wrote it when the test was trying (and failing) to bulk-import.  We get equivalent coverage from the export/import test now

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java (hunk 4 above)
Comment (labkey-chrisj) [reply src/org/labkey/test/tests/FileAttachmentColumnTest.java]: OK, I will modify the test accordingly

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 6]
```diff
@@ -15,153 +15,628 @@
...
+        for (File file : OTHER_SAMPLE_FILES)
+        {
+            uploadHelper.uploadFile(file);
+        }
+
+        // generate some resultFiles
+        var resultFiles = new ArrayList<File>();
+
+
+        String runFileContent = """
```
Comment (XingY) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java:251]: I meant that we could just use a file from the existing testAutomation/data folder so we don't have to generate those temp files

**Review by XingY — APPROVED** [review]

**Pull request merged**